ls1intum/Artemis
# `Development`: Switch from icon library to explicit reference

### Checklist
#### General


#### Client

### Motivation and Context


Change the icon usage from the icon library approach to the explicit reference approach.
The explicit reference approach involves explicitly importing the icon definition from the npm package, assigning it to the component's property and then binding this property to the icon input of the fa-icon component.

This approach is more verbose than the icon library approach, it makes it much easier to figure out where or whether the given icon is used. For example, with Find usages feature of the IDE. Another benefit of this approach is that it will produce a compile-time error if an icon is missing.

Also, this helps to further reduce the bundle size as icons are only loaded when needed.
### Description

Changed all icon imports and usages.

### Steps for Testing

Prerequisites:
???

1. Log in to Artemis
2. Click trough all screens (or as much as possible) on Artemis which has icons (almost all of them have)
3. Check that the icons are displayed
4. Have the dev-tools open and check if an error is displayed in the console

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage






### Screenshots
